WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryViewModel.swift

```diff
@@ -141,34 +151,67 @@ final class UpdateProductInventoryViewModel: ObservableObject {
         inventoryItem.imageURL
     }
 
-    func onTapIncreaseStockQuantityOnce() async {
+    func onTapIncreaseStockQuantityOnce() async throws {
         guard let quantityDecimal = Decimal(string: quantity) else {
             return
         }
 
         let newQuantity = quantityDecimal + 1
         quantity = newQuantity.formatted()
 
-        try? await updateStockQuantity(with: newQuantity)
+        do {
+            try await updateStockQuantity(with: newQuantity)
+        } catch {
+            throw UpdateInventoryError.generic
+        }
     }
 
-    func onTapUpdateStockQuantity() async {
+    func onTapUpdateStockQuantity() async throws {
         guard let quantityDecimal = Decimal(string: quantity) else {
-            return
+            throw UpdateInventoryError.nonSupportedQuantity
         }
 
-        try? await updateStockQuantity(with: quantityDecimal)
+        do {
+            try await updateStockQuantity(with: quantityDecimal)
+        } catch {
+            throw UpdateInventoryError.generic
+        }
+    }
+
+    func displayErrorNotice(_ productName: String) {
+        notice =  Notice(title: Localization.errorNoticetitle,
+                         message: String.localizedStringWithFormat(Localization.errorNoticeMessage, productName),
+                         feedbackType: .error)
     }
 }
 
 private extension UpdateProductInventoryViewModel {
     func updateStockQuantity(with newQuantity: Decimal) async throws {
         isPrimaryButtonLoading = true
 
-        // TODO: Handle error
-        try? await inventoryItem.updateStockQuantity(with: newQuantity, stores: stores)
+        do {
+            try await inventoryItem.updateStockQuantity(with: newQuantity, stores: stores)
+            onUpdatedInventory(newQuantity.formatted())
+        } catch {
+            isPrimaryButtonLoading = false
+            throw UpdateInventoryError.generic
```

Comment:
Here we throw the error, we don't need to duplicate it in other functions

Reply:
I see! Updated on https://github.com/woocommerce/woocommerce-ios/pull/11411/commits/e881252cedecdec6c61b1af88b6e86a7f74ae0fd